apps/server/src/modules/oauth-provider/controller/oauth-provider.controller.ts

```diff
@@ -24,67 +28,97 @@
 		private readonly oauthProviderResponseMapper: OauthProviderResponseMapper
 	) {}
 
+	private readonly defaultOauthClientBody: OauthClientBody = {
+		scope: 'openid offline',
+		grant_types: ['authorization_code', 'refresh_token'],
+		response_types: ['code', 'token', 'id_token'],
+		redirect_uris: [],
+	};
+
+	@RequireRole(RoleName.SUPERHERO)
 	@Authenticate('jwt')
 	@Get('clients/:id')
-	getOAuth2Client(@Param() params: IdParams) {
-		throw new NotImplementedException();
+	async getOAuth2Client(@Param() params: IdParams): Promise<OauthClientResponse> {
+		const client: OauthClient = await this.oauthProviderUc.getOAuth2Client(params.id);
+		const mapped: OauthClientResponse = this.oauthProviderResponseMapper.mapOauthClientToClientResponse(client);
+		return mapped;
 	}
 
+	@RequireRole(RoleName.SUPERHERO)
 	@Authenticate('jwt')
 	@Get('clients')
-	listOAuth2Clients(@Param() params: ListOauthClientsParams) {
-		throw new NotImplementedException();
+	async listOAuth2Clients(@Param() params: ListOauthClientsParams): Promise<OauthClientResponse[]> {
+		const clients: OauthClient[] = await this.oauthProviderUc.listOAuth2Clients(
+			params.limit,
+			params.offset,
+			params.client_name,
+			params.owner
+		);
+		const mapped: OauthClientResponse[] = clients.map(
+			(client: OauthClient): OauthClientResponse =>
+				this.oauthProviderResponseMapper.mapOauthClientToClientResponse(client)
+		);
+		return mapped;
 	}
 
+	@RequireRole(RoleName.SUPERHERO)
 	@Authenticate('jwt')
 	@Post('clients')
-	createOAuth2Client(@Body() body: OauthClientBody) {
-		throw new NotImplementedException();
+	async createOAuth2Client(@Body() body: OauthClientBody): Promise<OauthClientResponse> {
+		const bodyWithDefaults: OauthClientBody = { ...this.defaultOauthClientBody, ...body };
+		const client: OauthClient = await this.oauthProviderUc.createOAuth2Client(bodyWithDefaults);
+		const mapped: OauthClientResponse = this.oauthProviderResponseMapper.mapOauthClientToClientResponse(client);
+		return mapped;
 	}
 
+	@RequireRole(RoleName.SUPERHERO)
 	@Authenticate('jwt')
 	@Put('clients/:id')
-	updateOAuth2Client(@Param() params: IdParams, @Body() body: OauthClientBody) {
-		throw new NotImplementedException();
+	async updateOAuth2Client(@Param() params: IdParams, @Body() body: OauthClientBody): Promise<OauthClientResponse> {
+		const bodyWithDefaults: OauthClientBody = { ...this.defaultOauthClientBody, ...body };
```

Comment:
Mapping like this should be done in the UC rather than the Controller. This includes the default definition.